python/ngen_cal/src/ngen/cal/agent.py

```diff
@@ -57,9 +57,13 @@ def best_params(self) -> str:
 
 class Agent(BaseAgent):
 
-    def __init__(self, model_conf, workdir: 'Path', log: bool=False, restart: bool=False, parameters: 'Optional[Mapping[str, Any]]' = {}):
+    def __init__(self, model: Model, workdir: 'Path', log: bool=False, restart: bool=False, parameters: 'Optional[Mapping[str, Any]]' = {}):
         self._workdir = workdir
         self._job = None
+        assert not isinstance(model.model, NoModel), "invariant"
+        # NOTE: if support for new models is added, this will need to be modified
```

Comment:
It isn't clear here exactly what will need to be modified under this Note

Reply:
Just that `model.model` could likely be more than `Ngen | NoModel`, so getting the `unwrap` call would likely need to change. Do you want me to add something about that?